### PR TITLE
cli: migrate "tag list" to template

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Comment by jonathantanmy** [comment]
Looks good to me, but I haven't contributed to jj in a while so I'll let someone else approve.

This PR seems to do 3 things:
- add templating support to `tag list`
- add several more functions that a template can use
- use those functions to make a ref describer called `format_ref_targets`, and use `format_ref_targets` in the default template of `tag list`

These all seem reasonable to me and are very clearly laid out in this series of commits. I can't judge if the functions are what the jj project wants long-term (whether naming or functionality) so I'll let someone else approve.

**Review by noahmayr — APPROVED** [review]
lgtm as well, looking forward to the follow up for branch list templates

**Comment by yuja** [comment]
Thanks. I'll merge this PR after the release. This wouldn't be that useful without branch templating.

**Pull request merged**
